modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1645,6 +1645,94 @@ CV__DNN_INLINE_NS_BEGIN
                              float confThreshold = 0.5f, float nmsThreshold = 0.0f);
      };
 
+    /** @brief This class represents high-level API for YOLO object detection networks.
+      *
+      * YoloDetection allows to set params for preprocessing input image.
+      * YoloDetection creates net from file with trained weights and config,
+      * sets preprocessing input, runs forward pass and return result detections.
+      */
+     class CV_EXPORTS_W_SIMPLE YOLODetectionModel : public DetectionModel
+     {
+     public:
+        /**
+          * @brief Create detection model from network represented in one of the supported formats.
+          * An order of @p model and @p config arguments does not matter.
+          * @param[in] model Binary file contains trained weights.
+          * @param[in] config Text file contains network configuration.
+          * @param[in] version Version of the yolo model.
+          */
+         CV_WRAP YOLODetectionModel(const String& model, const String& config);
+
+         /**
+          * @brief Create model from onnx graph.
+          * @param[in] onnx path to onnx graph.
+          * @param[in] version Version of the yolo model.
+          */
+         CV_WRAP YOLODetectionModel(const String& onnx);
+
+         CV_DEPRECATED_EXTERNAL  // avoid using in C++ code (need to fix bindings first)
+         YOLODetectionModel();
+
+         /** @brief Given the @p input frame, create input blob, run net and return result detections.
+          *  @param[in]  frame  The input image.
+          *  @param[out] classIds Class indexes in result detection.
+          *  @param[out] confidences A set of corresponding confidences.
+          *  @param[out] boxes A set of bounding boxes.
+          *  @param[in] confThreshold A threshold used to filter boxes by confidences.
+          *  @param[in] nmsThreshold A threshold used in non maximum suppression.
+          */
+         CV_WRAP void detect(InputArray frame, CV_OUT std::vector<int>& classIds,
+                             CV_OUT std::vector<float>& confidences, CV_OUT std::vector<Rect>& boxes,
+                             float confThreshold = 0.5f, float nmsThreshold = 0.0f);
```

Comment:
Should be marked as `CV_OVERRIDE`

Reply:
PImpl public API doesn't use virtual or override methods.